pkg/agent/packetcapture/packetcapture_controller_test.go

```diff
@@ -0,0 +1,597 @@
+	packetCaptureInformer := crdInformerFactory.Crd().V1alpha1().PacketCaptures()
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
+
+	ifaceStore := interfacestore.NewInterfaceStore()
+	addPodInterface(ifaceStore, pod1.Namespace, pod1.Name, []string{pod1IPv4, ipv6}, pod1MAC.String(), int32(ofPortPod1))
+	addPodInterface(ifaceStore, pod2.Namespace, pod2.Name, []string{pod2IPv4}, pod2MAC.String(), int32(ofPortPod2))
+
+	// NewPacketCaptureController dont work on windows
+	pcController, err := NewPacketCaptureController(kubeClient, crdClient, packetCaptureInformer, ifaceStore)
+	if err != nil {
+		pcController = &Controller{
+			kubeClient:            kubeClient,
+			crdClient:             crdClient,
+			packetCaptureInformer: packetCaptureInformer,
+			packetCaptureLister:   packetCaptureInformer.Lister(),
+			packetCaptureSynced:   packetCaptureInformer.Informer().HasSynced,
+			interfaceStore:        ifaceStore,
+			captures:              make(map[string]*packetCaptureState),
+		}
+		packetCaptureInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+			AddFunc:    pcController.addPacketCapture,
+			UpdateFunc: pcController.updatePacketCapture,
+			DeleteFunc: pcController.deletePacketCapture,
+		}, resyncPeriod)
+	}
+
+	pcController.sftpUploader = &testUploader{}
+	pcController.captureInterface = &testCapture{}
+	pcController.queue = workqueue.NewTypedRateLimitingQueueWithConfig(
+		workqueue.NewTypedItemExponentialFailureRateLimiter[string](time.Millisecond*50, time.Millisecond*200),
+		workqueue.TypedRateLimitingQueueConfig[string]{Name: "packetcapture"},
+	)
+
+	t.Setenv("POD_NAME", "antrea-agent")
+	t.Setenv("POD_NAMESPACE", "kube-system")
+	return &fakePacketCaptureController{
+		Controller:         pcController,
+		kubeClient:         kubeClient,
+		mockController:     controller,
+		crdClient:          crdClient,
+		crdInformerFactory: crdInformerFactory,
+		informerFactory:    informerFactory,
+	}
+}
+
+func addPodInterface(ifaceStore interfacestore.InterfaceStore, podNamespace, podName string, podIPs []string, podMac string, ofPort int32) {
+	containerName := k8s.NamespacedName(podNamespace, podName)
+	var ifIPs []net.IP
+	for _, ip := range podIPs {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	mac, _ := net.ParseMAC(podMac)
+	ifaceStore.AddInterface(&interfacestore.InterfaceConfig{
+		IPs:                      ifIPs,
+		MAC:                      mac,
+		InterfaceName:            util.GenerateContainerInterfaceName(podName, podNamespace, containerName),
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{PodName: podName, PodNamespace: podNamespace, ContainerID: containerName},
+		OVSPortConfig:            &interfacestore.OVSPortConfig{OFPort: ofPort},
+	})
+}
+
+func TestMultiplePacketCaptures(t *testing.T) {
+	defaultFS = afero.NewMemMapFs()
+	packetsDir := "/tmp/antrea/packetcapture/packets"
+	defaultFS.MkdirAll(packetsDir, 0755)
```

Comment:
- we usually restore `defaultFS` at the end of the test
- I still don't understand why we need to create the directory as I think the `Run` method of the controller does it?
- we hardcode `/tmp` here, which is not equal to `os.TempDir()` on macOS; but because of the previous bullet point I am not sure it matters

Reply:
updated.